public/app/features/variables/editor/reducer.ts

```diff
@@ -1,7 +1,10 @@
+import { DataSourceApi } from '@grafana/data';
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { VariablePayload } from '../state/types';
+import { VariableQueryEditorType } from '../types';
 
 type VariableEditorExtension<ExtendedProps extends {} = {}> = { [P in keyof ExtendedProps]: ExtendedProps[P] };
```

Comment:
What's the point of this type? It's not exported to be used elsewhere, and it don't really seem to do much? Or am I missing something?

Reply:
The VariableEditorExtension type is used by the editor reducer and is a way to extend the state for various variable types.

Reply:
I'm not sure if I follow. This particular type - `VariableEditorExtension` - doesn't look like it's exported or reused elsewhere other than in this same module where it just sets an object.

```typescript
type VariableEditorExtension<ExtendedProps extends {} = {}> = { [P in keyof ExtendedProps]: ExtendedProps[P] };

export interface VariableEditorState<ExtendedProps extends {} = {}> {
  extended: VariableEditorExtension<ExtendedProps> | null;
}
```

seems to be the same as 

```typescript
export interface VariableEditorState<ExtendedProps extends {} = {}> {
  extended: ExtendedProps | null;
}
```

maybe?

Reply:
If you want to remove the `VariableEditorExtension` type then you need to add the type constraint as well so something like this:
```ts
export interface VariableEditorState<ExtendedProps extends {} = {}> {
  id: string;
  name: string;
  errors: Record<string, string>;
  isValid: boolean;
  extended: { [P in keyof ExtendedProps]: ExtendedProps[P] } | null;
}
```

I just thought that type made it clearer to understand what it does.